### ISSUE TITLE
e2e test failure: Kubectl apply should apply a new configuration to an existing RC 

### ISSUE THREAD

**Issue by davidopp** [kubernetes/kubernetes#27521]
[Fail] [k8s.io] Kubectl client [AfterEach] [k8s.io] Kubectl apply should apply a new configuration to an existing RC 
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:230

Assigning to test-infra team to triage.

**Comment by bprashanth** [comment]
Looks like this timed out on ns deletion, which in turn timed out because there was a pod with a deletion timestamp set but it was sitting around holding the ns after the timestamp:

```
Jun 15 21:43:07.136: INFO: Waiting up to 1m0s for all nodes to be ready
STEP: Destroying namespace "e2e-tests-kubectl-2d59e" for this suite.
Jun 15 21:48:07.180: INFO: Pod e2e-tests-kubectl-2d59e redis-master-wwm7t on node e2e-gce-agent-pr-22-0-minion-group-u16p remains, has deletion timestamp 2016-06-15T21:43:37-07:00
Jun 15 21:48:07.181: INFO: Couldn't delete ns "e2e-tests-kubectl-2d59e": namespace e2e-tests-kubectl-2d59e was not deleted within limit: timed out waiting for the condition, pods remaining: [redis-master-wwm7t]


â€¢ Failure in Spec Teardown (AfterEach) [300.457 seconds]
[k8s.io] Kubectl client
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:660
  [k8s.io] Kubectl apply [AfterEach]
  /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:660
    should apply a new configuration to an existing RC
    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/kubectl.go:445

    Jun 15 21:48:07.181: Couldn't delete ns "e2e-tests-kubectl-2d59e": namespace e2e-tests-kubectl-2d59e was not deleted within limit: timed out waiting for the condition, pods remaining: [redis-master-wwm7t]

    /go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:230
```
And the rc manager deleted the pod
```
I0616 04:43:07.273656       6 replication_controller.go:403] Pod e2e-tests-kubectl-2d59e/redis-master-wwm7t deleted through k8s.io/kubernetes/pkg/controller/replication.(*ReplicationManager).updatePod, timestamp 2016-06-16T04:43:37Z, labels map[role:master app:redis kubectl.kubernetes.io/apply-test:ADDED].
```

But in fact the last mention we see in kubelet logs of that lingering pod is around the time of the deltion timestamp, but it didn't hard-delete it from apiserver:
https://pantheon.corp.google.com/storage/browser/kubernetes-jenkins/pr-logs/pull/27518/kubernetes-pull-build-test-e2e-gce/44941/artifacts/e2e-gce-agent-pr-22-0-minion-group-u16p/
```
I0616 04:43:07.144895    3519 kubelet.go:2522] SyncLoop (RECONCILE, "api"): "redis-master-wwm7t_e2e-tests-kubectl-2d59e(d1dd95d0-337c-11e6-a808-42010af00002)"
I0616 04:43:07.145599    3519 manager.go:422] Status for pod "redis-master-wwm7t_e2e-tests-kubectl-2d59e(d1dd95d0-337c-11e6-a808-42010af00002)" updated successfully: {status:{Phase:Pending Conditions:[{Type:Initialized Status:True LastProbeTime:{Time:{sec:0 nsec:0 loc:0x4473220}} LastTransitionTime:{Time:{sec:63601648987 nsec:0 loc:0x4473220}} Reason: Message:} {Type:Ready Status:False LastProbeTime:{Time:{sec:0 nsec:0 loc:0x4473220}} LastTransitionTime:{Time:{sec:63601648987 nsec:0 loc:0x4473220}} Reason:ContainersNotReady Message:containers with unready status: [redis-master]} {Type:PodScheduled Status:True LastProbeTime:{Time:{sec:0 nsec:0 loc:0x4473220}} LastTransitionTime:{Time:{sec:63601648987 nsec:0 loc:0x4473220}} Reason: Message:}] Message: Reason: HostIP:10.240.0.5 PodIP: StartTime:0xc82013fde0 InitContainerStatuses:[] ContainerStatuses:[{Name:redis-master State:{Waiting:0xc82013fda0 Running:<nil> Terminated:<nil>} LastTerminationState:{Waiting:<nil> Running:<nil> Terminated:<nil>} Ready:false RestartCount:0 Image:gcr.io/google_containers/redis:e2e ImageID: ContainerID:}]} version:1 podName:redis-master-wwm7t podNamespace:e2e-tests-kubectl-2d59e}
I0616 04:43:07.243369    3519 config.go:269] Setting pods for source api
I0616 04:43:07.243964    3519 kubelet.go:2516] SyncLoop (UPDATE, "api"): "redis-master-wwm7t_e2e-tests-kubectl-2d59e(d1dd95d0-337c-11e6-a808-42010af00002):DeletionTimestamp=2016-06-16T04:43:37Z"
```

So looks like the node never saw the deletion timestamp? 
@kubernetes/goog-node

**Comment by bprashanth** [comment]
This is probably https://github.com/kubernetes/kubernetes/issues/26290#issuecomment-226390671

**Comment by yujuhong** [comment]
Yes, the test only checks whether the configuration has been applied to the RC, and once that's done, it moves on to deleting the entire namespace. From kubelet's perspective, it receives a pod ADD request, followed by the UPDATE (with deletion timestamp set) immediately. The pod worker was still stuck at the ADD requst, trying to mount a secret volume with the missing secret. This should've been fixed with PR #27491. Closing this issue so it'd be more clear if new failure happens.
